Consul.Test/AgentTest.cs

```diff
@@ -569,6 +569,150 @@ public async Task Agent_MonitorJSON()
             }
         }
 
+        [Theory]
+        [InlineData("passing")]
+        [InlineData("warning")]
+        [InlineData("critical")]
+        public async Task Agent_GetLocalServiceHealth(string statusString)
+        {
+            var healthStatus = HealthStatus.Parse(statusString);
+            var svcID = KVTest.GenerateTestKeyName();
+            var registration = new AgentServiceRegistration
+            {
+                Name = svcID,
+                Tags = new[] { "bar", "baz" },
+                Port = 8000,
+                Checks = new[]
+                {
+                    new AgentServiceCheck
+                    {
+                        TTL = TimeSpan.FromSeconds(15),
+                        Status = HealthStatus.Passing,
+                    },
+                    new AgentServiceCheck
+                    {
+                        TTL = TimeSpan.FromSeconds(15),
+                        Status = healthStatus,
+                    }
+                }
+            };
+
+            await _client.Agent.ServiceRegister(registration);
+
+            var status = await _client.Agent.GetLocalServiceHealth(svcID);
+
+            Assert.Equal(healthStatus, status.Response[0].AggregatedStatus);
+        }
+
+        [Theory]
+        [InlineData("passing")]
+        [InlineData("warning")]
+        [InlineData("critical")]
+        public async Task Agent_GetLocalServiceHealthByID(string statusString)
+        {
+            var healthStatus = HealthStatus.Parse(statusString);
+            var svcID = KVTest.GenerateTestKeyName();
+            var registration = new AgentServiceRegistration
+            {
+                ID = svcID,
+                Name = svcID,
```

Comment:
You are using the same names for `ID` and `Name`, please use different names so we know that it actually queries the service by its ID.